odoo/addons/base/models/res_lang.py

```diff
@@ -253,6 +253,24 @@ def _lang_code_to_urlcode(self, code):
                 return urlc
         return self._lang_get(code).url_code
 
+    @api.model
+    def _get_enabled_lang_code(self):
+        """ Get the list of code enabled (aka installed and default) """
+        langs = self._get_available_lang()
+        return [lang[0] for lang in langs]
+
+    @api.model
+    @tools.ormcache('self.iso_code')
+    def _get_available_lang(self):
+        """
+        Return the installed languages and the default one (aka iso_code is only 2 letters)
+        as a list of (code, name) sorted by the active state and the name (default order of model)
+        """
+        langs = self.with_context(active_test=False).search([])
+        lang_4iso_desactivated = langs.filtered(lambda lang: not lang.active and len(lang.iso_code) > 2)
+        langs_enabled = langs - lang_4iso_desactivated
+        return list(zip(langs_enabled.mapped('code'), langs_enabled.mapped('display_name')))
```

Comment:
return a res.lang record set. 
So you can use this directly in the field domain doing id in _get_available_lang.ids
and for def _lang_get(self):, do the tricks with list(zip) there.